src/ndb/ndb_internal.c

```diff
@@ -0,0 +1,426 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Roman Gezikov <rgezikov@exafore.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+#include <string.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <libswiftnav/logging.h>
+#include "libsbp/piksi.h"
+#include "sbp.h"
+#include "ndb.h"
+#include "ndb_internal.h"
+
+#define NDB_THREAD_PRIORITY (LOWPRIO)
+static WORKING_AREA_CCM(ndb_thread_wa, 1756);
+static void ndb_service_thread(void*);
+
+u8 ndb_file_version[MAX_NDB_FILE_VERSION_LEN];
+u8 ndb_file_end_mark = 0xb6;
+
+static bool do_writes_to_flash = false;
```

Comment:
Right. Sorry, forgot to discuss with that. This is temporary implementation. In theory we can implement a function that opens a file, saves data from ram and changes this flag to true. However, I don't see any need for that. So, I see two options: compile time define (the decision is not to use those, right?) or weak functions for file access returning success of open/close/seek/write and zeros on read. Which option is to be taken?